src/main/java/org/zalando/nakadi/controller/CursorOperationsController.java

```diff
@@ -0,0 +1,196 @@
+
+    @RequestMapping(path = "/event-types/{eventTypeName}/cursors-lag", method = RequestMethod.POST)
+    public ResponseEntity<?> cursorsLag(@PathVariable("eventTypeName") final String eventTypeName,
+                                        @Valid @RequestBody final ValidListWrapper<Cursor> cursors,
+                                        final Client client) {
+        accessControl(eventTypeName, client);
+
+        final List<NakadiCursor> domainCursor = cursors.getList().stream()
+                .map(toNakadiCursor(eventTypeName))
+                .collect(Collectors.toList());
+
+        final List<NakadiCursorLag> lagResult = cursorOperationsService
+                .cursorsLag(eventTypeName, domainCursor);
+
+        final List<CursorLag> viewResult = lagResult.stream().map(this::toCursorLag)
+                .collect(Collectors.toList());
+
+        return status(OK).body(viewResult);
+    }
+
+    @ExceptionHandler(InvalidCursorOperation.class)
+    public ResponseEntity<?> invalidCursorOperation(final InvalidCursorOperation e,
+                                                          final NativeWebRequest request) {
+        LOG.debug("User provided invalid cursor for operation. Reason: " + e.getReason(), e);
+        return Responses.create(Problem.valueOf(MoreStatus.UNPROCESSABLE_ENTITY,
+                clientErrorMessage(e.getReason())), request);
+    }
+
+    private String clientErrorMessage(final InvalidCursorOperation.Reason reason) {
+        switch (reason) {
+            case INVERTED_TIMELINE_ORDER: return "Inverted timelines. Final cursor must correspond to a newer " +
+                    "timeline than initial cursor.";
+
+            case TIMELINE_NOT_FOUND: return "Timeline not found. It might happen in case the cursor refers to a " +
+                    "timeline that has already expired.";
+            case INVERTED_OFFSET_ORDER: return "Inverted offsets. Final cursor offsets must be newer than initial " +
+                    "cursor offsets";
+            case PARTITION_NOT_FOUND: return "Partition not found.";
+            case CURSORS_WITH_DIFFERENT_PARTITION: return "Cursors with different partition. Pairs of cursors should " +
+                    "have matching partitions.";
+            default: return unexpectedErrorReason(reason);
+        }
+    }
+
+    private void accessControl(final String eventTypeName, final Client client) {
+        final EventType eventType;
+        try {
+            eventType = eventTypeRepository.findByName(eventTypeName);
+            client.checkScopes(eventType.getReadScopes());
+        } catch (final InternalNakadiException e) {
+            throw new MyNakadiRuntimeException1("failed to get event type", e);
+        } catch (final NoSuchEventTypeException e) {
+            throw new NoEventTypeException(e.getMessage(), e);
+        }
+    }
+
+    private String unexpectedErrorReason(final InvalidCursorOperation.Reason reason) throws MyNakadiRuntimeException1 {
```

Comment:
@rcillo I meant that method says that it returns String, but actually it only generates exception.
That is why I think that line `return unexpectedErrorReason(reason);` leads to misunderstanding. 
After reading method and it's call My brain was broken.
I think that 
```
default:
    throw generateUnexpectedError(reason)
``` 
is much better. 
And from the other hand - why not to do
```
default:
    LOG.error("Unexpected invalid cursor operation reason " + reason);
    throw new MyNakadiRuntimeException1();
```
?